x-pack/plugins/uptime/public/components/monitor/synthetics/step_detail/__tests__/use_monitor_breadcrumbs.test.tsx

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ChromeBreadcrumb } from 'kibana/public';
+import React from 'react';
+import { Route } from 'react-router-dom';
+import { of } from 'rxjs';
+import { MountWithReduxProvider, mountWithRouter } from '../../../../../lib';
+import { KibanaContextProvider } from '../../../../../../../../../src/plugins/kibana_react/public';
+import { useMonitorBreadcrumb } from '../use_monitor_breadcrumb';
+import { OVERVIEW_ROUTE } from '../../../../../../common/constants';
+import { Ping } from '../../../../../../common/runtime_types/ping';
+import { JourneyState } from '../../../../../state/reducers/journey';
+
+describe('useMonitorBreadcrumbs', () => {
+  it('sets the given breadcrumbs', () => {
+    const [getBreadcrumbs, core] = mockCore();
+
+    const Component = () => {
+      useMonitorBreadcrumb({
+        activeStep: { monitor: { id: 'test-monitor' } } as Ping,
+        journey: { details: { timestamp: '2021-01-04T11:25:19.104Z' } } as JourneyState,
+      });
+      return <>Step Water Fall</>;
+    };
+
+    mountWithRouter(
+      <MountWithReduxProvider>
+        <KibanaContextProvider services={{ ...core }}>
```

Comment:
there is no standard here, each developer just uses everything they seems fit, it's has been on my todo list to simplify the helper we have in uptime 

`x-pack/plugins/uptime/public/lib/helper/helper_with_router.tsx
`

Reply:
Andrew was concerned about this on one of my PRs during the break. Should we make an issue to cover this? Perhaps I can take a look at it since I may have less context to work on some of the larger bugs popping up for 7.11. It would be great if the helper was not opinionated, working only with a specific testing library, but rather spit out a component to be rendered with whatever library we choose.

Reply:
Yes, sure, that would be very nice. Just create an issue and go ahead.

Reply:
@dominiqueclarke for wrapper ideally i think we should use https://enzymejs.github.io/enzyme/docs/api/shallow.html#arguments

wrappingComponent prop, that means that part will also not go into snapshot and most use cases can be covered by shallow rendering.

Reply:
Anyway, let's continue this discussion in proper issue.